### PR TITLE
use openssl to take  a decision in just one task

### PR DESCRIPTION
**Pull request by pp-ps**
Using a Python script requires to use Task Variables, printing out their values in tasks, then take an authorization decision later on in a dedicated task.

This is overly complex, as we can take a final decision in the initial task, then just call acme-tiny.py as the only Python script.

### CODE REVIEW

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -18,40 +18,38 @@ stages:
...
-          - task: PythonScript@0
-            name: CheckCertificateExpiration
-            condition: and(succeeded(), eq(${{parameters.force_renew_cert}}, False))
-            inputs:
-              scriptSource: filePath
-              scriptPath: check_certificate_expiry.py
-              arguments: --certificate certificate.pem --delta $(CERT_NAME_EXPIRE_SECONDS)
+                if [ ${{ parameters.force_renew_cert }} == "True" ]
+                then
+                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
```
Comment (uolter) [inline azure-pipelines.yaml:29]:
```suggestion
                  echo "Force renew certificate."
                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
```

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -18,40 +18,38 @@ stages:
...
+                  exit 0
+                fi
+                if ! az keyvault certificate download --vault-name '$(KEY_VAULT_NAME)' -n '$(KEY_VAULT_CERT_NAME)' -f certificate.pem >/dev/null
+                then
+                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
```
Comment (uolter) [inline azure-pipelines.yaml:34]:
```suggestion
                  echo "Certificate not found in key vault."
                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
```

File: azure-pipelines.yaml [hunk 3]
```diff
@@ -18,40 +18,38 @@ stages:
...
+                  exit 0
+                else
+                  if ! openssl x509 -checkend '$(CERT_NAME_EXPIRE_SECONDS)' -noout -in certificate.pem >/dev/null 2>&1
+                  then
+                    # certificate is expiring or is already expired
```
Comment (uolter) [inline azure-pipelines.yaml:39]:
```suggestion
                   echo "certificate is expiring or is already expired"
```

**Review by uolter — CHANGES_REQUESTED** [review]

File: azure-pipelines.yaml (hunk 3 above)
Comment (pp-ps) [reply azure-pipelines.yaml:41]: Commit 4d516a4 removes /dev/null output from openssl, making it print this info.

File: azure-pipelines.yaml (hunk 2 above)
Comment (pp-ps) [reply azure-pipelines.yaml:35]: Added in 4d516a4.

File: azure-pipelines.yaml (hunk 1 above)
Comment (pp-ps) [reply azure-pipelines.yaml:29]: Added in 4d516a4.

**Review by uolter — APPROVED** [review]

**Pull request merged**